src/modules/PackageSlideShow.rb

```diff
@@ -188,150 +214,203 @@ def UpdateTotalProgress(_silent_check)
+    # Not to be confused with DownloadStart.
     #
-    # @param [String] pkg_name    package name
-    # @param [String] pkg_size    package size in bytes
-    # @param [Boolean] deleting    Flag: deleting (true) or installing (false) package
+    # @param [String]  pkg_name      package name
+    # @param [String]  _pkg_location full path to a package
+    # @param [String]  _pkg_summary  package summary (short description)
+    # @param [Integer] _pkg_size     installed package size in bytes
+    # @param [Boolean] deleting      Flag: deleting (true) or installing (false) package?
     #
-    def SlideDisplayDone(pkg_name, pkg_size, deleting)
-      if deleting
-        @removed_packages += 1
-
-        @removed_packages_list << pkg_name if Mode.normal
-      else
-        @total_size_installed += pkg_size
+    def PkgInstallStart(pkg_name, _pkg_location, _pkg_summary, _pkg_size, deleting)
+      if @active_downloads > 0 && !@detected_parallel_download
+        @detected_parallel_download = true
+        log.info "Detected parallel download and installation"
+      end
 
-        UpdateTotalProgress(false)
+      @updating = Pkg.PkgInstalled(pkg_name) unless deleting
+      UpdateTotalProgressText()
 
-        # Update global progress bar
-        DisplayGlobalProgress()
-
-        if @updating
-          @updated_packages = Ops.add(@updated_packages, 1)
-
-          if Mode.normal
-            @updated_packages_list = Builtins.add(
-              @updated_packages_list,
-              pkg_name
-            )
-          end
-        else
-          @installed_packages = Ops.add(@installed_packages, 1)
-
-          if Mode.normal
-            @installed_packages_list = Builtins.add(
-              @installed_packages_list,
-              pkg_name
-            )
-          end
-        end
-
-        @total_installed = Ops.add(@total_installed, pkg_size)
-      end
+      # Don't update the progress value since it cannot have changed right now:
+      # Only fully installed packages are taken into account, and this one has
+      # just begun.
 
       nil
     end
 
-    # package start display update
-    # - this is called at the beginning of a new package
+    # Progress notification while a package is finished being installed, updated or removed.
     #
-    # @param [String] pkg_name    package name
-    # @param [String] _pkg_location  full path to a package
-    # @param [String] _pkg_summary  package summary (short description)
-    # @param [Integer] _pkg_size    package size in bytes
-    # @param [Boolean] deleting    Flag: deleting (true) or installing (false) package?
+    # @param [Integer] _pkg_percent percent of progress of this package
     #
-    def SlideDisplayStart(pkg_name, _pkg_location, _pkg_summary, _pkg_size, deleting)
-      @updating = Pkg.PkgInstalled(pkg_name) if !deleting
-      # Update global progress bar
-      DisplayGlobalProgress()
-
+    def PkgInstallProgress(_pkg_percent)
```

Comment:
Coud this method be removed?

Reply:
Could, yes, at the cost of clarity since the libzypp callbacks one call level above do have such a method. This is still here for future use and for orientation of somebody reading the code later who will for sure wonder why this mirrors so many of the other callbacks (from the `SlideShowCallbacks` level), but not this one.

But you are right, this deserves a comment in the code.

Reply:
Added an explanatory comment.